src/datasources/asset/components/AssetQueryEditorCommon.tsx

```diff
@@ -0,0 +1,46 @@
+import { AssetDataSource } from "../AssetDataSource";
+import { AssetQuery } from "../types";
+import { CoreApp, QueryEditorProps, SelectableValue } from "@grafana/data";
+import { SystemMetadata } from "../../system/types";
+import { getVariableOptions } from "../../../core/utils";
+
+export type Props = QueryEditorProps<AssetDataSource, AssetQuery>
+
+export class AssetQueryEditorCommon {
```

Comment:
This is kind of a strange implementation here. you are basically having a common class that you use in 2 places. You are also creating references to your methods from the datasource and components . I think a better solution here is to create a custom react hook and pass the methods in the components here needed.